### ISSUE TITLE
Backend, problem with multiple "select2"

### ISSUE THREAD

**Issue by oeN** [solidusio/solidus#1996]
### Steps to reproduce
- Add a filter with a select box (with the select2 class) in the products index
- Click on the "New Product" button

### Expected behavior
- filters panel is hidden
- Shipping Category and Tax Category are styled correctly with **select2**

### Actual behavior
- filters panel is still visible
- Shipping Category and Tax Category are not styled

### System configuration
```
ruby 2.4

gem 'solidus', github: 'solidusio/solidus', branch: 'master'
```

**Extensions in use**:
```
gem 'solidus_auth_devise'
gem 'solidus_i18n', github: 'solidusio-contrib/solidus_i18n', branch: 'master'
gem 'globalize', github: 'globalize/globalize'
gem 'solidus_social', github: 'solidusio-contrib/solidus_social'
gem 'solidus_editor', github: 'solidusio-contrib/solidus_editor', branch: 'master'
gem 'solidus_paypal_braintree', github: 'solidusio/solidus_paypal_braintree'
gem 'solidus_globalize', github: 'solidusio-contrib/solidus_globalize'
```

**Comment by oeN** [comment]
The problem is related to https://github.com/select2/select2/issues/2750

We need to change the way solidus init the `select2()`. 
To solve this specific problem is enough to override the `backend/app/views/spree/admin/products/new.js.erb` and change `$('.select2').select2();` to `$('select.select2').select2();`

I am changing all the points where solidus uses `$('.select2').select2();` to `$('select.select2').select2();` and then I'll open a PR

**Comment by jhawthorn** [comment]
Fixed by #1997
